runtime/vm/jfr.cpp

```diff
@@ -125,6 +127,13 @@ static bool
 writeOutGlobalBuffer(J9VMThread *currentThread, bool finalWrite)
 {
 	J9JavaVM *vm = currentThread->javaVM;
+
```

Comment:
Check to make sure its false before you acquire the lock. You can keep the check inside the lock.